Yank/yank.py

```diff
@@ -473,6 +472,12 @@ def create(self, thermodynamic_state, sampler_states, topography, protocol,
             Simulation metadata to be stored in the file.
 
         """
+        # Check that protocol has same number of states for each parameter.
+        len_protocol_parameters = {par_name: len(path) for par_name, path in protocol.items()}
+        if len(set(len_protocol_parameters.values())) != 1:
+            raise ValueError('The protocol parameters have a different number '
+                             'of states: {}'.format(len_protocol_parameters))
```

Comment:
Yeah, it looks something like `{lambda_sterics: 22, lambda_electrostatics: 20}`. Not the prettiest, but it's a very short dict, so it should be readable.